src/modules/flow/iio/nodes.c

```diff
@@ -1571,7 +1452,7 @@ proximity_tick(struct sol_flow_node *node, void *data, uint16_t port, uint16_t c
         if (!sol_iio_device_trigger_now(mdata->device))
             goto error;
     } else {
-        proximity_reader_cb(node, mdata->device);
+        reader_cb(node, mdata->device);
     }
```

Comment:
Note that all `tick` and `close` methods are equal: you may define a `common_tick` and `common_close` and set nodes to use them on json.